src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/TraceLoggingEventSource.cs

```diff
@@ -151,6 +153,9 @@ public unsafe void Write(string? eventName, EventSourceOptions options)
         /// public instance properties of data will be written recursively to
         /// create the fields of the event.
         /// </param>
+#if !ES_BUILD_STANDALONE
+        [RequiresUnreferencedCode(EventSourceRequiresUnreferenceMessage)]
+#endif
         public unsafe void Write<T>(
```

Comment:
We could put a [`DynamicallyAccessedMembers(PublicProperties)`] on the T. That way our guidance could be "This can be suppressed if the object is a primitive type or a struct or class with primitive-typed public properties". It feels like it might cover a lot of scenarios.